src/apps/api/views/competitions.py

```diff
@@ -201,6 +209,19 @@ def email_all_participants(self, request, pk):
         batch_send_email.apply_async((comp.pk, content))
         return Response({}, status=status.HTTP_200_OK)
 
+    @action(detail=True, methods=['GET',],)
+    def get_csv(self, request, pk,):
+        buffer = StringIO.StringIO()
+        z = zipfile.ZipFile( buffer, "w")
```

Comment:
if you use "with" contexts, no need to worry about `z.close()`!

Kinda like this..

```
with zipfile.Zipfile(buffer, "w") as z:
    z.write("idletest")

# implicitly by leaving "with" context, it closes that file descriptor!
```